### PR TITLE
feat: SwiftUI app

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**

### CODE REVIEW

**Review by adrien-coye — COMMENTED** [review]
I did a first quick review. There will be some conflicts with my PR, but nothing that cannot be solved IMO.

File: Mail/Views/Thread/WebView.swift [hunk 1]
```diff
@@ -63,7 +63,7 @@ struct WebView: UIViewRepresentable {
         ) {
             if let url = navigationAction.request.url, Constants.isMailTo(url) {
                 decisionHandler(.cancel)
-                (parent.window?.windowScene?.delegate as? SceneDelegate)?.handleUrlOpen(url)
+                parent.openUrl.callAsFunction(url)
```
Comment (valentinperignon) [inline Mail/Views/Thread/WebView.swift:66]: Is it possible to write parent.openUrl(url)?
Just like the environment property dismiss

Comment (PhilippeWeidmann) [reply Mail/Views/Thread/WebView.swift:66]: It is

File: Mail/Views/Onboarding/OnboardingView.swift [hunk 2]
```diff
@@ -127,10 +126,9 @@ class LoginHandler: InfomaniakLoginDelegate, ObservableObject {
         Task {
             do {
                 _ = try await AccountManager.instance.createAndSetCurrentAccount(code: code, codeVerifier: verifier)
-                sceneDelegate?.showMainView()
                 UIApplication.shared.registerForRemoteNotifications()
             } catch let error as MailError where error == MailError.noMailbox {
-                sceneDelegate?.showNoMailboxView()
+                //sceneDelegate?.showNoMailboxView()
```
Comment (valentinperignon) [inline Mail/Views/Onboarding/OnboardingView.swift:131]: Will it be done in another PR?

File: Mail/Views/Switch User/AccountListView.swift [hunk 3]
```diff
@@ -79,7 +79,7 @@ struct AccountListView: View {
             isShowingNewAccountView = true
         }
         .fullScreenCover(isPresented: $isShowingNewAccountView, onDismiss: {
-            AppDelegate.orientationLock = .all
+           // AppDelegate.orientationLock = .all
```
Comment (valentinperignon) [inline Mail/Views/Switch User/AccountListView.swift:82]: Will it be done in another PR? Should be removed?

**Review by valentinperignon — COMMENTED** [review]
👍🏻

**Review by adrien-coye — DISMISSED** [review]
I re reviewed more in details and found nothing special to say. 
I also played with the app a bit, no issues there.
Let me know when it gets merged please 🙏

**Review by valentinperignon — APPROVED** [review]
Run perf tests before merging
